### ISSUE TITLE
destroySoon doesn't actually destroy the socket

### ISSUE THREAD

**Issue by dumbasPL** [postalsys/imapflow#200]
NodeJS's `destroySoon` will only destroy the socket once it's finished. Otherwise, it calls `.end()` and waits for the `finish` event which never happens in our case. This is problematic because after calling `.logout()` or `.close()` further calls to other APIs like `.idle()` don't error out and just silently fail (problematic if called in a loop for example). The current check for `if (this.socket.destroyed)` inside `.run()` will never trigger because the socket is never destroyed, just ended. Consider moving to `destroy()` or calling `finish` to actually destroy the socket.

temporary fix/hack:
```js
this.client.logout().then(() => (this.client as unknown as {socket: Socket}).socket.destroy())
```

**Comment by dumbasPL** [comment]
not stale

**Comment by andris9** [comment]
Fixed in v1.0.163
